lib/cdo/firehose.rb

```diff
@@ -43,63 +57,88 @@ class FirehoseClient < Cdo::Buffer
   # Ref: https://docs.aws.amazon.com/firehose/latest/dev/limits.html
   TRANSACTIONS_PER_SECOND = 2000.0
 
-  # Initializes the @firehose to an AWS Firehose client.
-  def initialize(log: CDO.log)
-    super(
-      batch_count: ITEMS_PER_REQUEST,
-      batch_size: BYTES_PER_REQUEST,
-      max_interval: 10.0,
-      min_interval: 1.0 / (TRANSACTIONS_PER_SECOND / Concurrent.processor_count),
-      log: log,
-      wait_at_exit: 10.0
-    )
+  class Buffer < Cdo::Buffer
+    # Initializes the @firehose to an AWS Firehose client.
+    def initialize(stream_name: nil)
+      super(
+        batch_count: ITEMS_PER_REQUEST,
+        batch_size: BYTES_PER_REQUEST,
+        max_interval: 10.0,
+        min_interval: 1.0 / (TRANSACTIONS_PER_SECOND / Concurrent.processor_count),
+        log: FirehoseClient.log,
+        wait_at_exit: 10.0
+      )
+
+      raise ArgumentError.new("stream_name must be defined") if stream_name.nil? || stream_name.blank?
```

Comment:
You can make a keyword argument required (and even better, caught at compile-time) by removing the default value, e.g.:

```ruby
def initialize(stream_name:)
```